Add cluster-api solution (#78)
open-cluster-management-io/OCM: solutions/cluster-api/README.md

tooling to simplify provisioning, upgrading, and operating multiple Kubernetes clusters. This doc is a guideline on how
to use `cluster-api` and `open-cluster-management` project together

## Prerequisite
Download and install [clusteradm](https://github.com/open-cluster-management-io/clusteradm/releases). For Linux OS, run the following commands:

```shell
wget -qO- https://github.com/open-cluster-management-io/clusteradm/releases/latest/download/clusteradm_linux_amd64.tar.gz | sudo tar -xvz -C /usr/local/bin/

sudo chmod +x /usr/local/bin/clusteradm
```

Follow the instruction [here](https://cluster-api.sigs.k8s.io/user/quick-start.html) to install the `clusterctl`

## Initialize the cluster api management plane and create cluster

Before initilize the management plane, some feature gates should be enabled:

```shell
export CLUSTER_TOPOLOGY=true
export EXP_CLUSTER_RESOURCE_SET=true
```

Next initiate the cluster api management plane by following [this](https://cluster-api.sigs.k8s.io/user/quick-start.html#initialize-the-management-cluster)

Then you can create a cluster on any cloud using cluster-api. Suppose the name of the cluster created is `capi-cluster`

> ## Initialize the ocm hub and register the cluster created by cluster api

Reviewer: ```suggestion
## Initialize the OCM multicluster control plane hub cluster and register the newly created cluster
```

> ocm hub can run in the same cluster of cluster api management plane

Reviewer: ```suggestion
The OCM multicluster control plane hub can be run in the same cluster as the Cluster API management plane. Use the following command to initialize the hub cluster: 
```

```
clusteradm init --use-bootstrap-token
```

> then run

Reviewer: ```suggestion
Register the newly created cluster by running the following command:
```

```

> ./capi-join.sh capi-cluster

Reviewer: ```suggestion
./register-join.sh <cluster-name> # ./register-join.sh capi-cluster
```

```

> 

Reviewer: ```suggestion

The registration process might take a while. After it's done, you can run the following command to verify the newly created cluster has successfully joined the hub cluster:

`kubectl get managedcluster`
```
